includes/REST_API/Stories_Page_Template_Controller.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Class Stories_Page_Template_Controller
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\REST_API;
+
+use stdClass;
+use WP_Error;
+use WP_Post;
+use WP_REST_Posts_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+/**
+ * Class Stories_Page_Template_Controller
+ *
+ * @package Google\Web_Stories\REST_API
+ */
+class Stories_Page_Template_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Constructor.
+	 *
+	 * Override the namespace.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $post_type Post type.
+	 */
+	public function __construct( $post_type ) {
+		parent::__construct( $post_type );
+		$this->namespace = 'web-stories/v1';
+	}
+
+	/**
+	 * Prepares a single story for create or update. Add post_content_filtered field to save/insert.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_post = parent::prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_post ) ) {
+			return $prepared_post;
+		}
+
+		// If the request is updating the content as well, let's make sure the JSON representation of the story is saved, too.
+		if ( isset( $request['page_data'] ) ) {
```

Comment:
Let's use `story_data` for consistency

Reply:
Why? That is really confusing, this is an a story, is a page, this is page data. I also think it makes the api confusing, as story_data is used elsewhere and for different data.

Reply:
A page template is exactly like a story, but with only one page in it, but contains the same story data like `version` and things like that.

Reply:
Fixed.